### PR TITLE
python user interface design

### PR DESCRIPTION
**Pull request by Superjom**
introduce some concepts

resolves: https://github.com/PaddlePaddle/Paddle/issues/3652

### CODE REVIEW

File: doc/design/python/user_interface.md [hunk 1]
```diff
@@ -0,0 +1,358 @@
+# User Interface Design
+
+## Basic Concepts
+### Variable
+A `Variable` represents shared, persistent state manipulated by a Paddle model program.
```
Comment (jacquesqiao) [inline doc/design/python/user_interface.md]: model program  ==> program?

File: doc/design/python/user_interface.md [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+
+Variables are maintained by `pd.Variable` class,
+each `pd.Variable` represents a tensor whose value can be changed by running ops on it.
```
Comment (jacquesqiao) [inline doc/design/python/user_interface.md]: running op on it is strange. maybe `changed by ops`

File: doc/design/python/user_interface.md [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+
+To make it more converient to share a variable, each `pd.Variable` has a name, 
+one can use a name to get or create a `pd.Variable` by calling `pd.get_variable`, for example:
+
+```python
+# same as 
+v = pd.get_variable(name="v", shape=[20, 20])
+```
+
+By default, Variables are model parameters, and will be updated after the network's back propagation.
```
Comment (jacquesqiao) [inline doc/design/python/user_interface.md]: seems we only use python Variable as parameters.

File: doc/design/python/user_interface.md [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+
+One can freeze a variable by setting `trainable` to `False` like:
+
+```python
+v = pd.Variable(shape=[20,20], trainable=False)
```
Comment (jacquesqiao) [inline doc/design/python/user_interface.md]: or change the state after creating it
```python
v = pd.Variable(...)
v.trianable = false
```

File: doc/design/python/user_interface.md [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+
+```
+v = pd.Variable(shape=[20,20], initializer=pd.zero_initializer())
+z = pd.Variable(shape=[20,20], initializer=pd.gaussian_initializer(mean=0., std=0.1))
+```
+
+to get the value of the variable, one can call
+
+```python
+print v.val()
```
Comment (jacquesqiao) [inline doc/design/python/user_interface.md]: maybe eval() is better

File: doc/design/python/user_interface.md (hunk 3 above)
Comment (Superjom) [reply doc/design/python/user_interface.md]: and placeholder

File: doc/design/python/user_interface.md (hunk 4 above)
Comment (Superjom) [reply doc/design/python/user_interface.md]: currently trainable is a member, so free to change all the arguments as member

File: doc/design/python/user_interface.md (hunk 5 above)
Comment (Superjom) [reply doc/design/python/user_interface.md]: this method just return a python numpy value, will not execute a sub-graph.

variable.eval() will be a wrapper based on `pd.eval`, currently is not necessary.

**Pull request closed without merging**